### DISCUSSION TITLE
wayland环境下，chromium没有提示框

### DISCUSSION THREAD

**Discussion by l631197874** [fcitx/fcitx5#684] [Q&A]
在WAYLANG环境下，火狐是好的，chromium没有提示框，chromium是由xwayland实现的。这边分析了下日志，发现提示框没有出现原因是xcbwindow没有update ui,没有打 Resize: 31 38这个日志，这边怀疑是不是xcb没有初始化成功？这边可以提供下分析的方向或者方法吗？系统的启动顺序是fcitx启动在lightdm后

**Comment by l631197874** [comment]
![243bbe621e0978c35fef3a8f4ff4fcb](https://user-images.githubusercontent.com/58616982/207208172-34e8873a-2574-45e7-a42e-f617a5159caf.jpg)

**Reply by l631197874** [reply]
上面是我kill后正常显示输入框打的日志，下面是出问题没有输入框的日志

**Reply by wengxt** [reply]
请贴fcitx5-diagnose 的输出

**Reply by lk-me** [reply]
# 系统信息:
1.  `uname -a`:

        Linux uos-PC 5.10.0-amd64-desktop #23.00.00.12-alpha SMP Thu Sep 15 16:36:05 CST 2022 x86_64 GNU/Linux

2.  `lsb_release -a`:

        No LSB modules are available.
        Distributor ID:	Uos
        Description:	UnionTech OS Desktop 23
        Release:	23
        Codename:	snipe

3.  `lsb_release -d`:

        Description:	UnionTech OS Desktop 23

4.  `/etc/lsb-release`:

        DISTRIB_ID=uos
        DISTRIB_RELEASE=20
        DISTRIB_DESCRIPTION="UnionTech OS 20"
        DISTRIB_CODENAME=eagle

5.  `/etc/os-release`:

        PRETTY_NAME="UnionTech OS Desktop 23"
        NAME="uos"
        VERSION_ID="23"
        VERSION="23"
        ID=uos
        HOME_URL="https://www.chinauos.com/"
        BUG_REPORT_URL="http://bbs.chinauos.com"
        VERSION_CODENAME=snipe

6.  桌面环境：

    桌面环境为 `deepin`。

7.  XDG 会话类型：

        XDG_SESSION_TYPE='wayland'

8.  Bash 版本：

        BASH_VERSION='5.1.8(1)-release'

# 环境：
1.  DISPLAY:

        DISPLAY=':0'


        WAYLAND_DISPLAY='wayland-0'

2.  键盘布局：

    1.  `setxkbmap`:

            xkb_keymap {
            	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
            	xkb_types     { include "complete"	};
            	xkb_compat    { include "complete"	};
            	xkb_symbols   { include "pc+us+inet(evdev)"	};
            	xkb_geometry  { include "pc(pc105)"	};
            };

    2.  `xprop`:

            _XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""

3.  Locale：

    1.  全部可用 locale：

            C
            C.UTF-8
            en_US.utf8
            POSIX
            zh_CN
            zh_CN.gb18030
            zh_CN.gb2312
            zh_CN.gbk
            zh_CN.utf8

    2.  当前 locale：

            LANG=zh_CN.UTF-8
            LC_CTYPE="zh_CN.UTF-8"
            LC_NUMERIC="zh_CN.UTF-8"
            LC_TIME="zh_CN.UTF-8"
            LC_COLLATE="zh_CN.UTF-8"
            LC_MONETARY="zh_CN.UTF-8"
            LC_MESSAGES="zh_CN.UTF-8"
            LC_PAPER="zh_CN.UTF-8"
            LC_NAME="zh_CN.UTF-8"
            LC_ADDRESS="zh_CN.UTF-8"
            LC_TELEPHONE="zh_CN.UTF-8"
            LC_MEASUREMENT="zh_CN.UTF-8"
            LC_IDENTIFICATION="zh_CN.UTF-8"
            LC_ALL=

4.  目录：

    1.  主目录：

            /home/uos

    2.  `${XDG_CONFIG_HOME}`:

        环境变量 `XDG_CONFIG_HOME` 是设定为 `/home/uos/.config` 。

        `XDG_CONFIG_HOME` 的当前值是 `~/.config` (`/home/uos/.config`)。

    3.  Fcitx5 设置目录：

        当前 fcitx5 设置目录是 `~/.config/fcitx5` (`/home/uos/.config/fcitx5`)。

5.  当前用户：

    脚本作为 uos (1000) 运行。

# Fcitx 状态:
1.  可执行文件：

    在 `/usr/bin/fcitx5` 找到了 fcitx5。

2.  版本：

    Fcitx 版本: `5.0.15`

3.  进程：

    找到了 1 个 fcitx5 进程：

           7418 fcitx5

4.  `fcitx5-remote`:

    `fcitx5-remote` 工作正常。

5.  DBus 界面：

    使用 `dbus-send` 来检查 dbus。

    DBus 名称 `org.fcitx.Fcitx5` 的所有者是 `:1.115`。

    DBus 名称 `org.fcitx.Fcitx5` 的 PID 所有者是 `7418`。

    来自 dbus 的调试信息：

           Group [x11::0] has 1 InputContext(s)
          IC [3ccdf5850fc54552a5cab15683f4dc3d] program:browser frontend:dbus cap:6000000012 focus:0
        Input Context without group
          IC [44a0e306cc6d45e9ab71c33dbb5c7daf] program:deepin-terminal frontend:dbus focus:1
          IC [902442de6cde422182ffa570bd4f8a44] program:dde-launcher frontend:dbus focus:0
          IC [9853b935fa9e48af82969272f975e779] program:dde-desktop frontend:dbus focus:0
          IC [04046df0dcb84e67872fbaead193215f] program:dde-network-dialog frontend:dbus focus:0

# Fcitx 配置界面：
1.  配置工具封装：

    在 `/usr/bin/fcitx5-configtool` 找到了 fcitx5-configtool。

2.  Qt 的配置界面：

    在 `/usr/bin/fcitx5-config-qt` 找到了 `fcitx5-config-qt`。

3.  KDE 的配置界面：

    找到了 fcitx5 的 kcm 模块。

        kcm_fcitx5                      - 配置输入法

# 前端设置：
## Xim:
1.  `${XMODIFIERS}`:

    环境变量 XMODIFIERS 已经正确地设为了“@im=fcitx”。
    从环境变量中获取的 Xim 服务名称为 fcitx.

2.  根窗口上的 XIM_SERVERS：

    Xim 服务的名称与环境变量中设置的相同。

## Qt:
1.  qt4 - `${QT4_IM_MODULE}`:

    环境变量 QT_IM_MODULE 已经正确地设为了“fcitx”。

2.  qt5 - `${QT_IM_MODULE}`:

    环境变量 QT_IM_MODULE 已经正确地设为了“fcitx”。

3.  Qt 输入法模块文件：

    找到了 fcitx5 qt5 模块：`/lib/x86_64-linux-gnu/fcitx5/qt5/libfcitx-quickphrase-editor5.so`。
    找到了 fcitx5 的 qt5 输入法模块：`/lib/x86_64-linux-gnu/qt5/plugins/platforminputcontexts/libfcitx5platforminputcontextplugin.so`。
    找到了未知的 fcitx qt 模块：`/lib/x86_64-linux-gnu/qt5/plugins/kcms/kcm_fcitx5.so`。
    **无法找到 Qt4 的 fcitx5 输入法模块。**

## Gtk:
1.  gtk - `${GTK_IM_MODULE}`:

    环境变量 GTK_IM_MODULE 已经正确地设为了“fcitx”。

2.  `gtk-query-immodules`:

    1.  gtk 2:

        **无法找到 gtk 2 的 `gtk-query-immodules`。**

        **无法找到 gtk 2 的 fcitx5 输入法模块。**

    2.  gtk 3:

        **无法找到 gtk 3 的 `gtk-query-immodules`。**

        **无法找到 gtk 3 的 fcitx5 输入法模块。**

3.  Gtk 输入法模块缓存：

    1.  gtk 2:

        在 `/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules.cache` 找到了 gtk `2.24.33` 的输入法模块缓存。
        版本行：

            # Created by /usr/lib/x86_64-linux-gnu/libgtk2.0-0/gtk-query-immodules-2.0 from gtk+-2.24.33

        已找到 gtk `2.24.33` 的 fcitx5 输入法模块。

            "/usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules/im-fcitx5.so" 
            "fcitx" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 
            "fcitx5" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 

    2.  gtk 3:

        在 `/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules.cache` 找到了 gtk `3.24.30` 的输入法模块缓存。
        版本行：

            # Created by /usr/lib/x86_64-linux-gnu/libgtk-3-0/gtk-query-immodules-3.0 from gtk+-3.24.30

        已找到 gtk `3.24.30` 的 fcitx5 输入法模块。

            "/usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-fcitx5.so" 
            "fcitx" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 
            "fcitx5" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 

    3.  gtk 4:

        **无法找到 gtk 4 的输入法模块缓存**

        **无法在缓存中找到 gtk 4 的 fcitx5 输入法模块。**

4.  Gtk 输入法模块文件：

    1.  gtk 2:

        找到的全部 Gtk 2 输入法模块文件均存在。

    2.  gtk 3:

        找到的全部 Gtk 3 输入法模块文件均存在。

    3.  gtk 4:

        找到的全部 Gtk 4 输入法模块文件均存在。

# 配置:
## Fcitx 插件：
1.  插件配置文件目录：

    找到了 fcitx5 的插件配置目录：`/usr/share/fcitx5/addon`。

2.  插件列表：

    1.  找到了 19 个已启用的插件：

            Classic User Interface 5.0.15
            Clipboard 5.0.15
            DBus 5.0.15
            DBus Frontend 5.0.15
            Fcitx4 Frontend 5.0.15
            IBus Frontend 5.0.15
            Input method selector 5.0.15
            Keyboard 5.0.15
            KDE Input Method Panel 5.0.15
            Status Notifier 5.0.15
            Notification 5.0.15
            Pinyin 5.0.11
            Extra Pinyin functionality 5.0.11
            Punctuation 5.0.11
            Spell 5.0.15
            Table 5.0.11
            Unicode 5.0.15
            XCB 5.0.15
            X Input Method Frontend 5.0.15

    2.  找到了 0 个被禁用的插件：

3.  插件库: 

    所有插件所需的库都被找到。

4.  用户界面：

    找到了 2 个已启用的用户界面插件：

        Classic User Interface
        KDE Input Method Panel

## 输入法：
1.  `/home/uos/.config/fcitx5/profile`:

        [Groups/0]
        # Group Name
        Name=默认
        # Layout
        Default Layout=us
        # Default Input Method
        DefaultIM=pinyin

        [Groups/0/Items/0]
        # Name
        Name=keyboard-us
        # Layout
        Layout=

        [Groups/0/Items/1]
        # Name
        Name=pinyin
        # Layout
        Layout=

        [GroupOrder]
        0=默认

# 日志：
1.  `date`:

        2022年 12月 13日 星期二 10:39:50 CST

2.  `/home/uos/.config/fcitx5/crash.log`:

    `/home/uos/.config/fcitx5/crash.log` 未找到.

**警告：fcitx5-diagnose 的输出可能包含敏感信息，包括发行版名称，内核版本，正在运行的程序名称等。**

**尽管这些信息对于开发者诊断问题有帮助，请在公开发送到在线网站前检查并且根据需要移除的对应信息。**

**Comment by lk-me** [comment]
没有打印 
XIM basic init: 1

**Comment by lk-me** [comment]
发现问题了是im-config没有控制好wayland与fctix启动顺序导致的
